backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/IHUQueryBuilder.java

```diff
@@ -476,5 +476,9 @@ public interface IHUQueryBuilder
 	 */
 	IHUQueryBuilder setOnlyStockedProducts(final boolean onlyStockedProducts);
 
-
+	/**
+	 * Do not consider {@link I_M_HU#COLUMNNAME_C_BPartner_ID} in the query, so we do not filter based on this column at all
+	 * Otherwise, if {@link #addOnlyInBPartnerId(BPartnerId)} gets <code>null</code> we will search only for those records with {@link I_M_HU#COLUMNNAME_C_BPartner_ID} <code>null</code>.
+	 */
+	IHUQueryBuilder setIgnoreBPartnerColumn();
```

Comment:
why do we need this flag?
precisely which is the case we cannot achieve by onlyIfAssignedToBPartner and OnlyInBPartnerIds fields.

PS: I have the feeling we are overcomplicating things for future devs.

Reply:
When we have a Picked HU for which the BPartner is set, but I don't know that BPartner directly from the product (interceptor), so I want to entirely omit `M_HU.C_BPartner_ID` when fetching the records.
By default in `de.metas.storage.spi.hu.impl.HUStorageQuery#HUStorageQuery`, `OnlyIfAssignedToBPartner` is set to `false` and `addOnlyInBPartnerId` to null.
Therefore, in `de.metas.handlingunits.impl.HUQueryBuilder#createQueryFilter` we'll end up with a Set which is not empty, and we'll filter only for those record with `C_BPartner_ID IS NULL`.
![image](https://github.com/metasfresh/metasfresh/assets/83770997/128d5acb-8a6e-48b9-b609-d3ac70d01935)